### DISCUSSION TITLE
Asynchronous update of zustand state isn't being awaited properly

### DISCUSSION THREAD

**Discussion by MEBonham** [pmndrs/zustand#1030] [Q&A]
Minimal reproduction of the bug:
[https://github.com/MEBonham/min-zustand-bug-reproduction/tree/main/minreproduction](https://github.com/MEBonham/min-zustand-bug-reproduction/tree/main/minreproduction)

Essentially the await in App.js line 13 isn't awaiting. This means that the variable passed to my SweetAlert modal is not updated like it should be after asynchronous functions in my Zustand Store.

If this isn't an issue in zustand, what's the most elegant workaround?

**Comment by dai-shi** [comment]
Here's what you want, right?
https://codesandbox.io/s/focused-joana-x4wjnr?file=/src/App.js

- setTimeout doesn't return a promise.
- you need to get non-stale value from store in useEffect.
